src/syntree.rs

```diff
@@ -111,69 +159,124 @@ pub struct Identifier {
     name: String,
 }
 
+/// The different kinds of [Assignable]s.
+/// 
+/// The recursive structure means that `a[2].b(1).c(2, 3)` is evaluated to
+/// ```ignored
+/// Access(
+///     Index(
+///         Read(a), 2),
+///         Access(
+///             Call(
+///                 Read(b), [1]
+///             ),
+///             Call(
+///                 Read(c), [2, 3]
+///             )
+///         )
+///     )
+/// ```
 #[derive(Debug, Clone)]
 pub enum AssignableKind {
     Read(Identifier),
+    /// A function call.
     Call(Box<Assignable>, Vec<Expression>),
     Access(Box<Assignable>, Box<Assignable>),
     Index(Box<Assignable>, Box<Expression>),
 }
 
+/// Something that can be assigned to. The assignable value can be read if the
+/// assignable is in an expression. Contains any [AssignableKind].
+///
+/// Note that something like `a = b` will be evaluated to
+/// ```ignored
+/// Statement::Assignment(
+///     Assignable::Read(a),
+///     Expression::Get(Assignable::Read(b))
+/// )
+/// ```
 #[derive(Debug, Clone)]
 pub struct Assignable {
     span: Span,
     kind: AssignableKind,
 }
 
+/// The different kinds of [Expression]s.
+/// 
+/// Expressions are recursive and end in some kind of value. Values are
+/// expressions as well.
 #[derive(Debug, Clone)]
 pub enum ExpressionKind {
+    /// Read from an [Assignable]. Variables, function calls, modules, blobs,
+    /// lists and tuples end up here.
```

Comment:
+ Dict indexing

A bit confusing about `modules, blobs, lists and tuples` - maybe clarify with: `module accesses, blob fields, list indexing, dict indexing and tuple indexing`